Aspire Budgeting/Facilities/GoogleSheetsManager.swift

```diff
@@ -253,60 +313,21 @@ final class GoogleSheetsManager: ObservableObject {
     }
   }
   
-  private func createSheetsValueRangeFrom(amount: String,
-                                          date: Date,
-                                          category: Int,
-                                          account: Int,
-                                          transactionType: Int,
-                                          approvalType: Int) -> GTLRSheets_ValueRange {
-    
-    let dateFormatter = DateFormatter()
-    dateFormatter.dateStyle = .medium
-    dateFormatter.timeStyle = .none
-    
-    let sheetsValueRange = GTLRSheets_ValueRange()
-    sheetsValueRange.majorDimension = kGTLRSheets_ValueRange_MajorDimension_Rows
-    sheetsValueRange.range = "Transactions!B:H"
-    
-    var valuesToInsert = [String]()
-    valuesToInsert.append(dateFormatter.string(from: date))
-    
-    if transactionType == 0 {
-      valuesToInsert.append("")
-      valuesToInsert.append(amount)
-    } else {
-      valuesToInsert.append(amount)
-      valuesToInsert.append("")
-    }
-    
-    valuesToInsert.append(transactionCategories![category])
-    valuesToInsert.append(transactionAccounts![account])
-    valuesToInsert.append("Added from Aspire iOS app")
-    
-    guard let version = self.aspireVersion else {
-      fatalError("Aspire Version is nil")
-    }
+  func fetchAccountBalances(spreadsheet: File) {
+    os_log("Fetching Account Balances",
+           log: .sheetsManager, type: .default)
     
-    switch version {
-    case .twoEight:
-      if approvalType == 0 {
-        valuesToInsert.append("🆗")
-      } else {
-        valuesToInsert.append("⏺")
-      }
-      
-    case .three, .threeOne:
-      if approvalType == 0 {
-        valuesToInsert.append("✅")
-      } else {
-        valuesToInsert.append("🅿️")
+    let range = "Dashboard!B10:C"
+    fetchData(spreadsheet: spreadsheet, spreadsheetRange: range) { (valueRange) in
+      if let values = valueRange.values as? [[String]] {
+        self.accountBalancesMetadata = AccountBalancesMetadata(metadata: values)
       }
     }
-    
-    sheetsValueRange.values = [valuesToInsert]
-    return sheetsValueRange
   }
-  
+}
+
+// MARK: Writing to Google Sheets
+extension GoogleSheetsManager {
   func addTransaction(amount: String, date: Date, category: Int, account: Int, transactionType: Int, approvalType: Int, completion: @escaping (Bool) -> Void) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/e6c795dd72b19b851d3b62f7797c3a34894b15f1/Aspire Budgeting/Facilities/GoogleSheetsManager.swift#L331"></span>Function should have 5 parameters or less: it currently has 7<br />
<code>function_parameter_count</code> <code>GoogleSheetsManager.swift:331</code></td>
    </tr>
  </tbody>
</table>